Sprint 17

AndrewkaW opened the pull request.

AndrewkaW committed:
Перенес создание settings и player из Creator.kt в Di

AndrewkaW committed:
Перенес создание player из Creator.kt в Koin

AndrewkaW committed:
Перенес создание Search из Creator в Koin и прибрался за собой.

Anna-Malinina commented on app/src/main/java/com/practicum/playlistmaker/App.kt:
```diff
@@ -2,15 +2,26 @@ package com.practicum.playlistmaker
 class App : Application() {
 
-    lateinit var settingsInteractor: SettingsInteractor
+    //lateinit var settingsInteractor: SettingsInteractor
```
Можно удалить, если не используется)

Anna-Malinina commented on app/src/main/java/com/practicum/playlistmaker/data/search/network/RetrofitNetworkClient.kt:
```diff
@@ -5,35 +5,32 @@ import android.net.ConnectivityManager
-        .addConverterFactory(GsonConverterFactory.create())
-        .build()
-    private val itunesService = retrofit.create(ItunesApi::class.java)
+class RetrofitNetworkClient(private val context: Context, private val itunesService: ItunesApi) :
```
![image](https://github.com/AndrewkaW/PlaylistMaker/assets/129777932/e2207a2f-8e4d-4db9-b696-dfef3e8d186a)
Обычно вот так переносят

Anna-Malinina replied:
У HistoryStorageImpl тоже так сделать можно

Anna-Malinina commented on app/src/main/java/com/practicum/playlistmaker/App.kt:
```diff
@@ -2,15 +2,26 @@ package com.practicum.playlistmaker
 
 import android.app.Application
 import com.practicum.playlistmaker.creator.Creator
```
Этот импорт не дает собраться, так как Creator уже удален
![image](https://github.com/AndrewkaW/PlaylistMaker/assets/129777932/a4eaf861-2f7f-483f-88f5-2e119fb1633c)

Anna-Malinina replied:
Для проверки я его, конечно же, убрала. У себя тоже сделай)

Anna-Malinina commented on app/src/main/java/com/practicum/playlistmaker/domain/Constants.kt:
```diff
@@ -14,6 +16,7 @@ class Constants {
         const val DARK_THEME = "DARK_THEME"
 
         const val CLICK_ITEM_DELAY = 1000L
+        const val ITUNES_URL = "http://itunes.apple.com"
```
Не обязательно делать класс с companion object для констант, можно просто их сделать в файле переменными без ничего лишнего

Anna-Malinina replied:
Первый скрин декомпилированный байткод, когда в классе как у тебя, второй скрин ,когда константы просто в файле. Можно заметить, что по сути получаем один и тот же public final класс со статическими полями. Только в первом случае он даже проще. (на скрин не влезла часть с декомпилированным companion object)
![image](https://github.com/AndrewkaW/PlaylistMaker/assets/129777932/a936d105-68a7-4351-8484-1ea04fdb5ed7)
![image](https://github.com/AndrewkaW/PlaylistMaker/assets/129777932/5a0b9b93-1b91-4b0d-9276-20bef8ac897c)

Anna-Malinina replied:
Посмотреть самому можно так:
![image](https://github.com/AndrewkaW/PlaylistMaker/assets/129777932/3323f4bf-1aff-401a-8195-9b8bf24fd3f5)

Anna-Malinina commented on app/src/main/java/com/practicum/playlistmaker/ui/player/activity/PlayerActivity.kt:
```diff
@@ -3,46 +3,37 @@ package com.practicum.playlistmaker.ui.player.activity
+    private val playerViewModel: PlayerViewModel by viewModel()
 
     override fun onCreate(savedInstanceState: Bundle?) {
+        @Suppress("DEPRECATION") val track = intent.getSerializableExtra(TRACK) as Track
```
Вообще по-хорошему его надо во вью модели хранить, как у тебя было в прошлом спринте

Anna-Malinina commented on app/src/main/java/com/practicum/playlistmaker/ui/settings/activity/SettingsActivity.kt:
```diff
@@ -5,23 +5,24 @@ import androidx.appcompat.app.AppCompatActivity
-            this,
-            SettingsViewModelFactory(this)
-        )[SettingsViewModel::class.java]
+//        viewModel = ViewModelProvider(
```
Можно с чистой душой удалять

Anna-Malinina commented on app/src/main/java/com/practicum/playlistmaker/data/search/network/RetrofitNetworkClient.kt:
```diff
@@ -5,35 +5,32 @@ import android.net.ConnectivityManager
                 (capabilities.hasTransport(NetworkCapabilities.TRANSPORT_CELLULAR) ||
-                capabilities.hasTransport(NetworkCapabilities.TRANSPORT_WIFI) ||
-                capabilities.hasTransport(NetworkCapabilities.TRANSPORT_ETHERNET))
+                        capabilities.hasTransport(NetworkCapabilities.TRANSPORT_WIFI) ||
```
Здесь немного съехало, студия предлагает убрать эти отступы

Anna-Malinina reviewed (commented):
Забавный баг:  
https://github.com/AndrewkaW/PlaylistMaker/assets/129777932/70c92da1-2bee-4bd7-9f12-d0263e2a1104

Баг из крита:  

https://github.com/AndrewkaW/PlaylistMaker/assets/129777932/a7e5bb54-ff06-4334-9339-8fcbf427c0d8

AndrewkaW committed:
Фикс 3х багов:
С тайпером плеера при спаме кнопки плей.
С обновлением списка истории.
С поиском при очистке текста запроса.

AndrewkaW committed:
Правки по рекомендациям и замечаниям
